### PR TITLE
Dc analogs event detection enable

### PR DESCRIPTION
**Pull request by dcmaverick**
Added Event detection enable to AV/AI 
JIRA: https://jira.se.com/browse/FIRMWARE-29075

### CODE REVIEW

File: src/bacnet/basic/object/ai.c [hunk 1]
```diff
@@ -395,10 +396,11 @@ const char *Analog_Input_Name_ASCII(uint32_t object_instance)
 unsigned Analog_Input_Event_State(uint32_t object_instance)
 {
     unsigned state = EVENT_STATE_NORMAL;
-#if defined(INTRINSIC_REPORTING)
+#if !defined(INTRINSIC_REPORTING)
```
Comment (metelik) [inline src/bacnet/basic/object/ai.c:399]: Why the change of compilation logic?

**Review by metelik — COMMENTED** [review]
Please do not forget about setting it to true at the init i. e.: https://github.com/se-apc/bacnet-stack/blob/781caadc53eaa48ca8410231688a9f2c48b51ca7/src/bacnet/basic/object/bv.c#L1345 .

File: src/bacnet/basic/object/ai.c (hunk 1 above)
Comment (dcmaverick) [reply src/bacnet/basic/object/ai.c:399]: That looks like a mistake on my behalf

File: src/bacnet/basic/object/ai.c [hunk 2]
```diff
@@ -1190,6 +1245,10 @@ int Analog_Input_Read_Property(BACNET_READ_PROPERTY_DATA *rpdata)
                                                                  : false);
             apdu_len = encode_application_bitstring(&apdu[0], &bit_string);
             break;
+        case PROP_EVENT_DETECTION_ENABLE:
```
Comment (metelik) [inline src/bacnet/basic/object/ai.c:1248]: # if defined (INTRINSIC_REPORTING) ?

File: src/bacnet/basic/object/ai.c [hunk 3]
```diff
@@ -409,6 +411,59 @@ unsigned Analog_Input_Event_State(uint32_t object_instance)
...
+ * value
+ *
+ * @param  object_instance - object-instance number of the object
+ *
+ * @return  event-detection-enable property value
+ */
+bool Analog_Input_Event_Detection_Enable(uint32_t object_instance)
+{
+    bool retval = false;
+#if !defined(INTRINSIC_REPORTING)
```
Comment (metelik) [inline src/bacnet/basic/object/ai.c:426]: no need for conditional compilation here as it is covered by the line 414

File: src/bacnet/basic/object/ai.c [hunk 4]
```diff
@@ -409,6 +411,59 @@ unsigned Analog_Input_Event_State(uint32_t object_instance)
...
+ *
+ * @param  object_instance - object-instance number of the object
+ *
+ * @return  event-detection-enable property value
+ */
+bool Analog_Input_Event_Detection_Enable_Set(
+    uint32_t object_instance, bool value)
+{
+    bool retval = false;
+#if !defined(INTRINSIC_REPORTING)
```
Comment (metelik) [inline src/bacnet/basic/object/ai.c:451]: line 414 cnditional comp covers the logic

**Review by metelik — COMMENTED** [review]
More commentary ;)

File: src/bacnet/basic/object/ai.c [hunk 5]
```diff
@@ -45,6 +45,7 @@ static const int Properties_Optional[] = {
     PROP_TIME_DELAY, PROP_NOTIFICATION_CLASS, PROP_HIGH_LIMIT,
     PROP_LOW_LIMIT, PROP_DEADBAND, PROP_LIMIT_ENABLE, PROP_EVENT_ENABLE,
     PROP_ACKED_TRANSITIONS, PROP_NOTIFY_TYPE, PROP_EVENT_TIME_STAMPS,
+    PROP_EVENT_DETECTION_ENABLE,
```
Comment (metelik) [inline src/bacnet/basic/object/ai.c:48]: 5 These properties shall be present only if the object supports intrinsic reporting. ;)

**Review by metelik — COMMENTED** [review]
.

File: src/bacnet/basic/object/ai.c (hunk 2 above)
Comment (metelik) [reply src/bacnet/basic/object/ai.c:1248]: ok it is in line 1203

**Review by metelik — APPROVED** [review]
